app/iht/views/application/gift/given_away.scala.html

```diff
@@ -36,14 +36,17 @@
 
     @helpers.form(action = iht.controllers.application.gifts.routes.GivenAwayController.onSubmit,'autoComplete -> "off") {
 
-    <p>@Messages("page.iht.application.gifts.lastYears.givenAway.p1",getDateBeforeSevenYears(getOrException(regDetails.deceasedDateOfDeath).dateOfDeath).toString(dateFormatForDisplay),getOrException(regDetails.deceasedDateOfDeath).dateOfDeath.toString(dateFormatForDisplay))</p>
-    <p>@Messages("page.iht.application.gifts.lastYears.givenAway.p2")</p>
+    <p>@Messages("page.iht.application.gifts.lastYears.givenAway.p1",
+                        getDeceasedNameOrDefaultString(regDetails),
```

Comment:
Maybe prefix getDeceasedNameOrDefaultString and getDateBeforeSevenYears to make it clear where these methods are defined. You have done elsewhere with the first.